docs/interfaces.md

```diff
@@ -12,184 +12,349 @@ Slots may support multiple connections to plugs.  For example the OS snap
 exposes the ``network`` slot and all applications that can talk over the
 network connect their plugs there.
 
-## Supported Interfaces - Basic
+The availability of an interface depends on whether snapd is running on a
+classic (eg, traditional desktop or server) or on a native system. Interfaces
+may also be implicit to the OS snap or implemented only via snaps providing the
+slot.
 
-### network
+## Native vs classic interfaces
```

Comment:
That said, I reworded this section and it is now called 'Transitional interfaces'.